api/tests/e2e/e2e_suite_test.go

```diff
@@ -59,6 +67,16 @@ var _ = BeforeSuite(func() {
 	ensureServerIsUp()
 })
 
+var _ = BeforeEach(func() {
+	serviceAccountName = generateGUID("user")
+	token := obtainServiceAccountToken(serviceAccountName)
```

Comment:
`obtainServiceAccountToken` does a fair bit of work, so we're introducing a significant overhead into each test here.

Consider whether this is necessary for every e2e test. If so, it might make more sense to be shared in a synchronised before suite.